### PR TITLE
Add GNSS Library v3 and SD_MMC - in preparation for adding the reference_station

### PR DESCRIPTION
**Pull request by PaulZC**

### CODE REVIEW

File: Firmware/RTK_Surveyor/Begin.ino [hunk 1]
```diff
@@ -273,80 +274,111 @@ void beginSD()
...
     }
-
-    //Change to root directory. All new file creation will be in root.
-    if (sd->chdir() == false)
+    else
     {
-      systemPrintln("SD change directory failed");
-      break;
+      // SDIO MMC
+      if (SD_MMC.begin() == false)
```
Comment (nseidle) [inline Firmware/RTK_Surveyor/Begin.ino:368]: We use beginSD() and the sdPresent() function as a software card-detect (in absence of hardware CD). I believe you have hardware CD so please be sure the system fails gracefully if an SD is not present. Also please test that hot insert/removal is supported.

File: Firmware/RTK_Surveyor/System.ino [hunk 2]
```diff
@@ -293,18 +293,35 @@ bool isConnected(uint8_t deviceAddress)
...
+    //File successfully created
+    testFile.close();
+    if (sd->exists(testFileName))
+      sd->remove(testFileName);
+    return (!sd->exists(testFileName));
+  }
+  else
+  {
+    // SDIO MMC
+    char testFileName[40] = "/testfile.txt"; // Create it in the root directory
```
Comment (nseidle) [inline Firmware/RTK_Surveyor/System.ino]: I believe sdFat handles /fileName.ext just fine. You could consolidate some of the char[]s.

**Comment by nseidle** [comment]
This PR looks great to me!

File: Firmware/RTK_Surveyor/System.ino (hunk 2 above)
Comment (PaulZC) [reply Firmware/RTK_Surveyor/System.ino]: OK. I'll test this once I have functional Reference_Station code.
My plan at the moment is to have duplicate names for SdFat vs SD_MMC, without and with the slash. But if SdFat does ignore the slash, then we can use common names.
I'm adding this to the [sheet](https://docs.google.com/spreadsheets/d/1TNjSrNvyVTpTsuWSNNbTeBClg1CfebEc8Dg3rV33QRg/edit#gid=447334257) and will track it there.

**Comment by PaulZC** [comment]
Excellent. Merging...

**Comment by PaulZC** [comment]
@nseidle : when you get a moment, please have a look at the comments here:

https://github.com/sparkfun/SparkFun_RTK_Firmware/blob/832bd5b29f43510e4e9ab0b3834a50c8ea2a0bd6/Firmware/RTK_Surveyor/support.ino#L451-L457

I think line 457 should be ```if (data <= 3)``` ?

**Pull request merged**